src/java/voldemort/store/readonly/ReadOnlyFileDiffer.java

```diff
@@ -0,0 +1,383 @@
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import voldemort.utils.ByteUtils;
+import voldemort.utils.EventThrottler;
+import voldemort.utils.Pair;
+
+
+/**
+ * ReadOnlyFileDiffer is used to compare between different versions of specific store and give the number of how many
+ * records are added/deleted/updated/same.
+ *
+ * N.B.: This tool only works with the Read-Only V2 format.
+ */
+public class ReadOnlyFileDiffer {
+    public final static int ADD = 0;
+    public final static int DELETE = 1;
+    public final static int UPDATE = 2;
+    public final static int SAME = 3;
+
+    public static void main(String[] args) {
+        //Initialize options of command.
+        Options options = new Options();
+        Option pathOption = new Option("p", "path", true, "Path of Voldemort readonly data dir");
+        pathOption.setRequired(true);
+        Option storeOption = new Option("s", "store", true, "Store name");
+        storeOption.setRequired(true);
+        Option oldOption = new Option("o", "oldVersion", true, "Old version number");
+        oldOption.setRequired(true);
+        Option newOption = new Option("n", "newVersion", true, "New version number");
+        newOption.setRequired(true);
+        options.addOption(pathOption);
+        options.addOption(storeOption);
+        options.addOption(oldOption);
+        options.addOption(newOption);
+        options.addOption("r", "rate", true, "Max rate of reading bytes from disk per second.");
+        options.addOption("h", "help", false, "Print usage");
+
+        CommandLineParser parser = new BasicParser();
+        CommandLine cmd;
+        try {
+            cmd = parser.parse(options, args);
+            //Print help information.
+            if (cmd.hasOption("h")) {
+                HelpFormatter formatter = new HelpFormatter();
+                formatter.printHelp("ReadOnlyFileDiffer", options);
+                return;
+            }
+
+            String path = cmd.getOptionValue("p");
+            String store = cmd.getOptionValue("s");
+            int oldVersion = Integer.valueOf(cmd.getOptionValue("o"));
+            int newVersion = Integer.valueOf(cmd.getOptionValue("n"));
+            int rate = Integer.valueOf(cmd.getOptionValue("r", "-1"));
+
+            ReadOnlyFileDiffer analyzer = new ReadOnlyFileDiffer();
+
+            System.out.println(
+                "Start comparing for store:" + store + " between version-" + oldVersion + " and version-" + newVersion);
+
+            long[] diffResults = analyzer.compareForStore(path, store, oldVersion, newVersion, rate);
+
+            System.out.println(
+                "Diff of store" + store + ". Added:" + diffResults[ADD] + " Deleted:" + diffResults[DELETE]
+                    + " Updated:" + diffResults[UPDATE] + " Same:" + diffResults[SAME]);
+        } catch (ParseException e) {
+            System.err.println(e.getMessage());
+            System.exit(-1);
```

Comment:
why print log by system out rather than by logger? there is no time info printed by console